tests/test_stea.py

```diff
@@ -199,6 +203,68 @@ def test_units_and_scale_factor(
     )
 
 
+@pytest.mark.parametrize(
+    "start_year, end_year, expected_final_fopt, exception",
+    [
+        # The mocked EclSum object contains data every tenth date from
+        # 2010-01-01 and onwards for 1000 days. The 990th dah is 2012-09-17.
+        # There is no data for the 1000th day, so when summing FOPT, the FOPR
+        # for the last entry is not accounted for, thus we end with 990 as the
+        # final FOPT. FOPR is 1 for all days.
+        (None, None, 990, None),
+        (2010, None, 990, None),
+        (2000, 2013, 990, None),
+        (2010, 2013, 990, None),
+        (2011, 2013, 990 - 365, None),
+        (2012, 2013, 990 - 2 * 365, None),
+        (2013, 2013, None, ValueError),
+        (2010, 2012, 990, None),  # end-year includes all data up to the year after
+        (2010, 2011, 2 * 365, None),
+        (2010, 2010, 365, None),
+        (2010, 2009, None, ValueError),
+    ],
+)
+def test_start_year_end_year(
+    start_year, end_year, expected_final_fopt, exception, tmpdir, mock_project
+):
+    os.chdir(tmpdir)
+    config = {
+        SteaInputKeys.CONFIG_DATE: datetime.datetime(2018, 10, 10, 12, 0, 0),
+        SteaInputKeys.PROJECT_ID: 1234,
+        SteaInputKeys.PROJECT_VERSION: 1,
+        SteaInputKeys.ECL_PROFILES: {
+            "ID1": {
+                SteaInputKeys.ECL_KEY: "FOPT",
+                SteaInputKeys.START_YEAR: start_year,
+                SteaInputKeys.END_YEAR: end_year,  # NB: None here gives 'null' in yaml
+            }
+        },
+        SteaInputKeys.RESULTS: ["npv"],
+        SteaInputKeys.ECL_CASE: "CSV",
+    }
+    pathlib.Path("config_file").write_text(yaml.dump(config), encoding="utf-8")
+
+    # Mock ECL binary output files:
+    case: EclSum = create_case()
+    case.fwrite()
+
+    stea_input = SteaInput(["config_file"])
+
+    if exception is None:
+        request = make_request(stea_input, mock_project)
+        print(request.request_data["Adjustments"]["Profiles"][0]["Data"]["Data"])
+        assert (
+            pytest.approx(
+                sum(request.request_data["Adjustments"]["Profiles"][0]["Data"]["Data"])
+                * 1e6
+            )
+            == expected_final_fopt
+        )
+    else:
+        with pytest.raises(exception):
+            make_request(stea_input, mock_project)
+
+
```

Comment:
This is the recommended way of dealing with parameterized failing tests:

```suggestion
@pytest.mark.parametrize(
    "start_year, end_year, expected_final_fopt, expectation",
    [
        # The mocked EclSum object contains data every tenth date from
        # 2010-01-01 and onwards for 1000 days. The 990th dah is 2012-09-17.
        # There is no data for the 1000th day, so when summing FOPT, the FOPR
        # for the last entry is not accounted for, thus we end with 990 as the
        # final FOPT. FOPR is 1 for all days.
        (None, None, 990, does_not_raise()),
        (2010, None, 990, does_not_raise()),
        (2000, 2013, 990, does_not_raise()),
        (2010, 2013, 990, does_not_raise()),
        (2011, 2013, 990 - 365, does_not_raise()),
        (2012, 2013, 990 - 2 * 365, does_not_raise()),
        (2013, 2013, None, pytest.raises(ValueError, match="Invalid time interval start after end")),
        (2010, 2012, 990, does_not_raise()),  # end-year includes all data up to the year after
        (2010, 2011, 2 * 365, does_not_raise()),
        (2010, 2010, 365, does_not_raise()),
        (2010, 2009, None, pytest.raises(ValueError, match="Invalid time interval start after end")),
    ],
)
def test_start_year_end_year(
    start_year, end_year, expected_final_fopt, expectation, tmpdir, mock_project
):
    os.chdir(tmpdir)
    config = {
        SteaInputKeys.CONFIG_DATE: datetime.datetime(2018, 10, 10, 12, 0, 0),
        SteaInputKeys.PROJECT_ID: 1234,
        SteaInputKeys.PROJECT_VERSION: 1,
        SteaInputKeys.ECL_PROFILES: {
            "ID1": {
                SteaInputKeys.ECL_KEY: "FOPT",
                SteaInputKeys.START_YEAR: start_year,
                SteaInputKeys.END_YEAR: end_year,  # NB: None here gives 'null' in yaml
            }
        },
        SteaInputKeys.RESULTS: ["npv"],
        SteaInputKeys.ECL_CASE: "CSV",
    }
    pathlib.Path("config_file").write_text(yaml.dump(config), encoding="utf-8")

    # Mock ECL binary output files:
    case: EclSum = create_case()
    case.fwrite()

    stea_input = SteaInput(["config_file"])

    if expectation is None:
        request = make_request(stea_input, mock_project)
        assert (
            pytest.approx(
                sum(request.request_data["Adjustments"]["Profiles"][0]["Data"]["Data"])
                * 1e6
            )
            == expected_final_fopt
        )

```
also need: `from contextlib import ExitStack as does_not_raise`

Also, could you add an `id` to the different tests to make it clear what they are testing?

Reply:
Fixed